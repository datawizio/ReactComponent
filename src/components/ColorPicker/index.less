.color-picker-popover {
  width: 220px;
  .ant-popover-inner-content {
    padding: 0;
  }
}

.color-picker {
  width: 20px;
  height: 20px;
  border-radius: 3px;
  border: 1px solid rgb(217, 217, 217);
  cursor: pointer;
}
