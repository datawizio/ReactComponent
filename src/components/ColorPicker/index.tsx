import React from "react";
import { Popover } from "antd";
import { SketchPicker, ColorResult } from "react-color";

import "./index.less";

export interface ColorPickerProps {
  value?: string;
  onChange?: (value: string) => void;
  defaultColors?: string[];
}

const ColorPicker: React.FC<ColorPickerProps> = ({
  defaultColors,
  value,
  onChange
}) => {
  const handleChange = (val: ColorResult) => {
    onChange && onChange(val.hex);
  };
  return (
    <Popover
      trigger="click"
      overlayClassName="color-picker-popover"
      content={
        <SketchPicker
          color={value ?? "#FFF"}
          presetColors={defaultColors}
          disableAlpha
          onChange={handleChange}
        />
      }
    >
      <div className="color-picker" style={{ background: value }}></div>
    </Popover>
  );
};

export default ColorPicker;
