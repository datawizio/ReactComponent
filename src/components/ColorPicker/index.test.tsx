import "jsdom-global/register";
import React from "react";
import { mount, shallow } from "enzyme";

import ColorPicker from "./index";

window.HTMLCanvasElement.prototype.getContext = jest.fn();

const mockProps = {
  value: "#A771FE",
  onChange: jest.fn(),
  defaultColors: ["#782FEF", "#231F20", "#E8D5D5", "#DF8600", "#FBE7B5"],
  render: jest.fn()
};

const setUp = (props?) => shallow(<ColorPicker {...props} />);

describe("ColorPicker component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("ColorPicker with props rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("ColorPicker without props rendered correctly", () => {
    const wrapper = setUp();

    expect(wrapper).toMatchSnapshot();
  });

  it("ColorPicker popover open render corectly", async () => {
    const wrapper = shallow(<ColorPicker />);
    wrapper.simulate("click");
    expect(wrapper).toMatchSnapshot();
  });

  it.skip("ColorPicker onChange", () => {
    const mockedEvent = { target: { value: { hex: "#4d1d1d" } } };

    const wrapper = mount(<ColorPicker {...mockProps} />);
    wrapper.simulate("click");
    const popoverContent = wrapper.find(".ant-popover-inner-content");
    //@ts-ignore
    popoverContent.props().children.props.onChange(mockedEvent);
    expect(wrapper.prop("onChange")).toBeCalledTimes(1);
  });
});
