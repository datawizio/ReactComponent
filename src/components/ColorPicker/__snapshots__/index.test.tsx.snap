// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component ColorPicker onChange 1`] = `
<Tooltip
  arrowPointAtCenter={false}
  autoAdjustOverflow={true}
  destroyTooltipOnHide={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  overlay={
    <React.Fragment>
      <div
        className="ant-popover-inner-content"
      >
        <ColorPicker
          color="#A771FE"
          disableAlpha={true}
          onChange={[Function]}
          presetColors={
            Array [
              "#782FEF",
              "#231F20",
              "#E8D5D5",
              "#DF8600",
              "#FBE7B5",
            ]
          }
          styles={Object {}}
          width={200}
        />
      </div>
    </React.Fragment>
  }
  overlayClassName="color-picker-popover"
  overlayStyle={Object {}}
  placement="top"
  prefixCls="ant-popover"
  transitionName="zoom-big"
  trigger="click"
/>
`;

exports[`ColorPicker component ColorPicker popover open render corectly 1`] = `
<Popover
  content={
    <ColorPicker
      color="#FFF"
      disableAlpha={true}
      onChange={[Function]}
      presetColors={
        Array [
          "#D0021B",
          "#F5A623",
          "#F8E71C",
          "#8B572A",
          "#7ED321",
          "#417505",
          "#BD10E0",
          "#9013FE",
          "#4A90E2",
          "#50E3C2",
          "#B8E986",
          "#000000",
          "#4A4A4A",
          "#9B9B9B",
          "#FFFFFF",
        ]
      }
      styles={Object {}}
      width={200}
    />
  }
  destroyTooltipOnHide={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  overlayClassName="color-picker-popover"
  overlayStyle={Object {}}
  placement="top"
  transitionName="zoom-big"
  trigger="click"
>
  <div
    className="color-picker"
    style={
      Object {
        "background": undefined,
      }
    }
  />
</Popover>
`;

exports[`ColorPicker component ColorPicker with props rendered correctly 1`] = `
<Popover
  content={
    <ColorPicker
      color="#A771FE"
      disableAlpha={true}
      onChange={[Function]}
      presetColors={
        Array [
          "#782FEF",
          "#231F20",
          "#E8D5D5",
          "#DF8600",
          "#FBE7B5",
        ]
      }
      styles={Object {}}
      width={200}
    />
  }
  destroyTooltipOnHide={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  overlayClassName="color-picker-popover"
  overlayStyle={Object {}}
  placement="top"
  transitionName="zoom-big"
  trigger="click"
/>
`;

exports[`ColorPicker component ColorPicker without props rendered correctly 1`] = `
<Popover
  content={
    <ColorPicker
      color="#FFF"
      disableAlpha={true}
      onChange={[Function]}
      presetColors={
        Array [
          "#D0021B",
          "#F5A623",
          "#F8E71C",
          "#8B572A",
          "#7ED321",
          "#417505",
          "#BD10E0",
          "#9013FE",
          "#4A90E2",
          "#50E3C2",
          "#B8E986",
          "#000000",
          "#4A4A4A",
          "#9B9B9B",
          "#FFFFFF",
        ]
      }
      styles={Object {}}
      width={200}
    />
  }
  destroyTooltipOnHide={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  overlayClassName="color-picker-popover"
  overlayStyle={Object {}}
  placement="top"
  transitionName="zoom-big"
  trigger="click"
>
  <div
    className="color-picker"
    style={
      Object {
        "background": undefined,
      }
    }
  />
</Popover>
`;
